packages/hardhat/deploy/08_deploy_crab_v2.ts

```diff
@@ -0,0 +1,106 @@
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { createArgumentFile, getCrab, getDwethToken, getEuler, getExec, getUniswapDeployments, getWETH } from '../tasks/utils'
+import { getPoolAddress } from '../test/setup';
+import { one } from '../test/utils';
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts, ethers, network } = hre;
+  const { deploy } = deployments;
+
+  const { deployer } = await getNamedAccounts();
+
+  const controller = await ethers.getContract("Controller", deployer);
+  const oracle = await ethers.getContract("Oracle", deployer);
+  const weth = await getWETH(ethers, deployer, network.name)
+  const wsqueeth = await ethers.getContract("WPowerPerp", deployer);
+  const crabV1 = getCrab(network.name);
+
+  const { uniswapFactory, } = await getUniswapDeployments(ethers, deployer, network.name)
+
+  const squeethPoolAddr = await getPoolAddress(wsqueeth, weth, uniswapFactory)
+
+
+  console.log(controller.address)
+
+
+
+
+  const exec = await getExec(deployer, network.name)
+  const euler = await getEuler(deployer, network.name)
+  const dWethToken = await getDwethToken(deployer, network.name)
+
+  const migrationArgs = [
+    crabV1,
+    weth.address,
+    exec,
+    dWethToken,
+    euler,
+  ]
+
+  // Deploy CrabMigration contract
+  await deploy("CrabMigration", {
+    from: deployer,
+    log: true,
+    args: migrationArgs,
+    skipIfAlreadyDeployed: true
+  });
+
+  console.log(`Successfully deploy CrabMigration`)
+
+
+  const timelockArgs = [
+    '0x609FFF64429e2A275a879e5C50e415cec842c629', // deployer,
+    432000,
+  ]
+
+
+  // Deploy Timelock contract
+  await deploy("Timelock", {
+    from: deployer,
+    log: true,
+    args: timelockArgs,
+    skipIfAlreadyDeployed: true
+  });
+
+
+
+  console.log(`Successfully deploy Timelock`)
```

Comment:
how do you get the addresses of the contracts deployed? Do you want to print that out in the script as well?

Reply:
It will automatically print